### ISSUE TITLE
CSS review

### ISSUE THREAD

**Issue by laurelfulford** [WordPress/twentyseventeen#12]
The theme's CSS needs to be reviewed to make sure it follows best practices. 

This should be done once #4 is complete.

**Comment by Presskopp** [comment]
I wonder if these lines from L.2449 like
_.main-navigation ul ul ul a:hover,_
can be shortened using odd/even and the like (selectors). Or is this needed because of BC?
Sorry if I'm spamming.

**Comment by marcustisater** [comment]
I'm up for it! So just to make sure, are we following the [WordPress CSS Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/)?

**Comment by grappler** [comment]
Yes, we are following the coding standards.

**Comment by marcustisater** [comment]
Thanks man, I will have a look and leave PR.

**Comment by juanfra** [comment]
@marcustisater check this out: https://rezzz.com/php-codesniffer-with-wordpress-coding-standards-and-atom/ - It's a great tool to check that.

**Comment by celloexpressions** [comment]
I saw a number of minor issues when I went through the whole stylesheet for colors yesterday. There were a couple of duplicate selector blocks in there I think.

**Comment by marcustisater** [comment]
Thanks for the heads up, expect a PR in a couple of days 😄

**Comment by joyously** [comment]
One best practice is to keep the theme's selectors at a consistent specificity, so that the cascade makes sense and overrides are easier. Most selectors in this theme use classes, but a few are using IDs, which are more specific. 
`#page`
`#masthead .wrap`
`body:not(.twentyseventeen-front-page) #comments`
`.single-post:not(.has-sidebar) #primary, .page.page-one-column #primary,
.home.blog:not(.has-sidebar) #primary`
`#comments`
`#secondary`
`.twentyseventeen-front-page.twentyseventeen-customizer #primary article.panel-placeholder`
`.has-sidebar:not(.error404) #primary`
`.has-sidebar #secondary`
`.error404 #primary`
`body.has-sidebar.error404 #primary .page-header,
    body.page-two-column #primary .entry-header`
`.blog:not(.has-sidebar):not(.home) #primary article,
    .archive:not(.has-sidebar) #primary article,
    .search:not(.has-sidebar) #primary article,
    .error404:not(.has-sidebar) #primary .page-content,
    .error404.has-sidebar #primary .page-content,
    body.page-two-column #primary .entry-content`
`.has-sidebar #primary .entry-content blockquote.alignright.below-entry-meta`
`#comments` /\* again! _/
`#secondary` /_ again! */
Also, some of the selectors should be simplified as much as possible and still select what is wanted. For example, `.has-sidebar` instead of `body.has-sidebar`.
It is also best practice to use distinct names, but I see `button.secondary` and `#secondary`, which is confusing.

**Comment by marcustisater** [comment]
@joyously good point, I noticed some of these things myself when I went through the stylesheet. I think that's for another issue to keep this one more clean and referable.  

I started solving the selector properties with [PR](https://github.com/WordPress/twentyseventeen/pull/169):

**Comment by josephfusco** [comment]
`:before` & `:after` are deprecated in favor of `::before` & `::after` ([w3c](https://www.w3.org/TR/css3-selectors/#gen-content)) with double colons which allow for better semantics. Since it will affect IE8 and below, should we just be using the `::` instead and include the older single `:` in the IE specific stylesheets?

I see this often in themes as it is easier to just have `:` and save time and cut down on IE specific stylesheets, but this theme sets the example for many.

**Comment by marcustisater** [comment]
Ahaa, I didn't know that was deprecated, good catch. I agree with you. Would you mind making the change to the PR?

**Comment by davidakennedy** [comment]
Closing the loop on some issues from @marcustisater in another issue: https://github.com/WordPress/twentyseventeen/issues/170

These should be fixed.

**Comment by ntwb** [comment]
@josephfusco WordPress uses single colons for pseudo selectors for the most part even though this is undocumented in the standards.

**Comment by joyously** [comment]
Why are `.alignright` and `.alignleft` set to display:inline? It seems like that might mess up other styling such as top/bottom margins and padding if applied to a div in the post editor.

**Comment by davidakennedy** [comment]
Closing this, as the theme had several reviews. Of course, continued scrutiny when the theme merges to Core is welcome.
